Features/testing

vcherdantsev opened the pull request:
Task 4: 

- interfaces Repo (Check and Test) and Flusher (Check and Test) added
- generated mocks
- 3 tests added for TestFlusher

rusdevops commented on internal/flusher/flusher.go:
```diff
@@ -0,0 +1,56 @@
+
+	for i := 0; i < len(bulks); i = i + 1 {
+		if err := f.checkRepo.AddChecks(bulks[i]); err != nil {
+			return bulks[i]
```
Метод должен вернуть оставшуюся часть всех сущностей.

vcherdantsev replied:
Исправил, но rebase переписал коммит, к которому был привязан комментарий.

rusdevops commented on internal/flusher/flusher.go:
```diff
@@ -0,0 +1,56 @@
+
+	for i := 0; i < len(bulks); i = i + 1 {
+		if err := f.testRepo.AddTests(bulks[i]); err != nil {
+			return bulks[i]
```
Метод должен вернуть оставшуюся часть всех сущностей.

vcherdantsev replied:
Исправил, но rebase переписал коммит, к которому был привязан комментарий.

rusdevops commented on internal/models/test.go:
```diff
@@ -0,0 +1,23 @@
+	Output string `json:"output,omitempty"`
+}
+
+func NewTest(testID uint64, taskID uint64, input string, output string) (*Test, error) {
```
Попрошу убрать данную функцию

vcherdantsev replied:
Убрал, но rebase переписал коммит, к которому был привязан комментарий.

rusdevops commented on internal/utils/utils.go:
```diff
@@ -86,11 +86,29 @@ func SplitToBulks(checks []api.Check, batchSize uint) (batches [][]api.Check) {
 }
+
+func SplitTestsToBulks(tests []models.Test, batchSize uint) (batches [][]models.Test) {
+	for int(batchSize) < len(tests) {
```
попрошу добавить проверку для `batchSize == 0`

vcherdantsev replied:
Добавил.

Commit:
Testing frameworks added and interfaces.

Commit:
New test added. Some code refactoring.

Commit:
Mocks added.

Commit:
Changes after review:
- NewTest, NewCheck methods are removed
- Split* functions return error when batchSize is zero
- AddTask/AddCheck methods of Repo return correct remained slice of tests/checks.

rusdevops commented on internal/utils/utils.go:
```diff
@@ -95,12 +102,19 @@ func ConvertChecksToMap(checks []models.Check) (map[uint64]models.Check, error)
-func SplitTestsToBulks(tests []models.Test, batchSize uint) (batches [][]models.Test) {
+func SplitTestsToBulks(tests []models.Test, batchSize uint) (batches [][]models.Test, err error) {
+	if batchSize == 0 {
+		batches = nil
```
можно убрать ?

rusdevops commented on internal/utils/utils.go:
```diff
@@ -95,12 +102,19 @@ func ConvertChecksToMap(checks []models.Check) (map[uint64]models.Check, error)
 	}
 
 	batches = append(batches, tests)
+	err = nil
```
можно убрать ?

Commit:
Task 4: removed explicit nil in return values.
